services/agit/agit.go

```diff
@@ -46,6 +125,49 @@ func ProcReceive(ctx context.Context, repo *repo_model.Repository, gitRepo *git.
 			continue
 		}
 
+		if opts.RefFullNames[i].IsForReview() {
```

Comment:
This for-loop is too large to maintain. I think we need to split it into small functions (and it's easier to use unit test to cover the small functions)